extension/src/cli/dvc/discovery.ts

```diff
@@ -12,90 +8,82 @@ import {
   getConfigValue,
   setUserConfigValue
 } from '../../vscode/config'
-import { getPythonBinPath } from '../../extensions/python'
 import { getFirstWorkspaceFolder } from '../../vscode/workspaceFolders'
 import { delay } from '../../util/time'
 import { SetupSection } from '../../setup/webview/contract'
 
-export const warnUnableToVerifyVersion = () =>
-  Toast.warnWithOptions(
+const warnWithSetupAction = async (
+  setup: IExtensionSetup,
+  warningText: string
+): Promise<void> => {
+  const response = await Toast.warnWithOptions(warningText, Response.SHOW_SETUP)
+
+  if (response === Response.SHOW_SETUP) {
+    return setup.showSetup(SetupSection.DVC)
+  }
+}
+
+const warnUnableToVerifyVersion = (setup: IExtensionSetup) => {
+  void warnWithSetupAction(
+    setup,
     'The extension cannot initialize as we were unable to verify the DVC CLI version.'
   )
+}
 
-export const warnVersionIncompatible = (
-  version: string,
+const warnVersionIncompatible = (
+  setup: IExtensionSetup,
   update: 'CLI' | 'extension'
 ): void => {
-  void Toast.warnWithOptions(
-    `The extension cannot initialize because you are using version ${version} of the DVC CLI. The expected version is ${MIN_CLI_VERSION} <= DVC < ${MAX_CLI_VERSION}. Please upgrade to the most recent version of the ${update} and reload this window.`
+  void warnWithSetupAction(
+    setup,
+    `The extension cannot initialize because you are using the wrong version of the ${update}.`
```

Comment:
[C] Let's simplify this further and say something along the lines of the CLI is incompatible with this version of the extension. The user can open setup to get more details.

Reply:
maybe get closer to `'The extension cannot initialize as we were unable to verify the DVC CLI version.'` so `'The extension cannot initialize because the DVC CLI version is incompatible.'`